http/request.pony

```diff
@@ -0,0 +1,194 @@
+
+primitive HTTP11 is (Equatable[HTTPVersion] & Stringable)
+  """
+  HTTP/1.1
+  """
+  fun string(): String iso^ => recover iso String(8).>append("HTTP/1.1") end
+  fun to_bytes(): Array[U8] val => [as U8: 'H'; 'T'; 'T'; 'P'; '/'; '1'; '.'; '1']
+  fun eq(o: HTTPVersion): Bool => o is this
+
+primitive HTTP10 is (Equatable[HTTPVersion] & Stringable)
+  """
+  HTTP/1.0
+  """
+  fun string(): String iso^ => recover iso String(8).>append("HTTP/1.0") end
+  fun to_bytes(): Array[U8] val => [as U8: 'H'; 'T'; 'T'; 'P'; '/'; '1'; '.'; '0']
+  fun eq(o: HTTPVersion): Bool => o is this
+
+primitive HTTP09 is (Equatable[HTTPVersion] & Stringable)
+  """
+  HTTP/0.9
+  """
+  fun string(): String iso^ => recover iso String(8).>append("HTTP/0.9") end
+  fun to_bytes(): Array[U8] val => [as U8: 'H'; 'T'; 'T'; 'P'; '/'; '0'; '.'; '9']
+  fun eq(o: HTTPVersion): Bool => o is this
+
+type HTTPVersion is (HTTP09 | HTTP10 | HTTP11)
```

Comment:
It would be useful to allow this type to expose an interface, like this:
```pony
type HTTPVersion is ((HTTP09 | HTTP10 | HTTP11) & _HTTPVersion)

interface val _HTTPVersion is (Equatable[HTTPVersion] & Stringable)
  fun to_bytes(): Array[U8] val

primitive HTTP11 is _HTTPVersion
...
```